src/SmartNumber.php

```diff
@@ -270,34 +303,20 @@ public function getHumanString(bool $preferFraction = true): string
 	 */
 	public function setValue(string $value): void
 	{
-		$value = NumberHelper::preprocessInput($value);
-
-		// Store input
 		$this->input = $value;
+		$value = NumberHelper::preprocessInput($value, $this->decimalPointSigns ?: [], $this->thousandsSeparators ?: []);
 
 		if (Validators::isNumeric($value)) {
-			$toInteger = (string) preg_replace('/\..*$/', '', $value);
```

Comment:
Please keep this fixer for broken user input in case of `integer`.

Reply:
Could you please write me a concrete SmartNumber input, so I can write a test for it?

Reply:
Some inputs (by query log from Mathematicator database):

- `45`
- `10.000`
- `4.`
- `3.0`

All these inputs must be converted to an integer, because the integer is the canonical form.

Reply:
I'll solve it later today.

Reply:
I merge this PR and this suggestion can be solved in #9.